doc/questions.md

```diff
@@ -0,0 +1,53 @@
+## Questions
+
+[Questions][] are a mechanism that allows the user to react to certain situations that occur in the installer.
+
+[Questions]: https://opensuse.github.io/agama/dbus/ref-org.opensuse.Agama.Questions1.html
+
+### Questions and unattended installation
+
+Questions work well, but in some cases, such as automatic or CLI-driven installation, questions can be problematic. So we introduce a mechanism to make even questions
+non-blocking and non-interactive. The main goal is to give the user a flexible way of answering questions that he knows in advance.
+and for the rest, which are not specified or not yet seen, simply use the default answer.
+
+#### Requirements for Questions
+
+To make it work there are several requirements for questions:
+
+1. each question has a hierarchical ID to allow partial matching. For example `storage.luks.password` or
+software.repository.checksum. Why: it allows in the future when questions are consistent enough to allow something like
+storage.* -> :skip.
+2. every question has a defined default secure answer. Why: it allows real unattended mode even if user does not specify all questions.
+3. questions can define additional params in map that help user to answer them. Example repository url and its checksum.
+   Partition identification for luks password. Why: Getting data only from question text is hard and very unreadable with regexp.
+   So questions need to specify some easy to match data in addition to the pure text.
+   E.g. checksum and repository url for automatic repository approval.
+4. Question can work in attended and unattended mode. Why: To know when to apply default answer. See 2.
+
+#### Unattended Installation Features
+
+1. Questions contain values in addition to ids that can be matched. To match
+id and values, string, regexp or array can be used. Array lists all possible values and regexp is used for matching.
```

Comment:
IMHO regexp (regular expressions) are overkill (incompatible dialects...) and I hope substrings or shell patterns will be enough

Reply:
that is good suggestion. Maybe for the first phase we can omit regexp completely and if needed, we can extend it ( so some string that start with some fancy character to support advanced matching ).